### PR TITLE
Added product name to homepage & thumbnail sizing

### LINKED ISSUE #2229: Add product names to homepage
Based on design in https://github.com/mozilla/foundation.mozilla.org/issues/2191
Blocked by https://github.com/mozilla/foundation.mozilla.org/issues/2124

Note:
- company name is .h6-heading and product name is .body but #909090
- names are hidden on mobile

Comps: https://redpen.io/p/qz6bdb2811e058ffb1

red-pen-dfeeef57-446d-46b1-f151-13508e67836c

red-pen-4a7b925a-9a9d-43bc-efe3-85f5750f7960

image

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #2229 and #2124 

@Pomax I added a few wrappers so it will be a lot easier if you review the code without whitespace changes: https://github.com/mozilla/foundation.mozilla.org/pull/2232/files?w=1

### CODE REVIEW

**Comment by kristinashu** [comment]
This is looking great! Feels so nice and responsive! Few things, let me know if you have any questions.

- [ ] Add white cutout 

I noticed this white cutout is missing. Screen cap from PR
screen shot 2018-11-20 at 10 44 22 am

Screen cap from prod:
screen shot 2018-11-20 at 10 43 42 am

- [ ] Please remove the white space at the bottom so product images go right up to the footer

Screen cap from PR:
screen shot 2018-11-20 at 10 46 03 am

Suggested comp:
screen shot 2018-11-20 at 10 46 03 am

- [ ] Add a state with four images across

As the screen width narrrows, it goes from 5 across, to 3, and then to 2. Would it be possible for it to go from 5 across, to 4, to 3 and then to 2? Because just before it goes from 5 to 3, the images are quite small. I think added 4 across will help with this.

File: source/sass/buyers-guide/views/homepage.scss [hunk 1]
```diff
@@ -1,4 +1,8 @@
 #pni-home {
+  .main-content.pb-5 {
+    padding-bottom: 0 !important;
```
[the remark this answers is not in the document]
Comment (Pomax) [reply source/sass/buyers-guide/views/homepage.scss]: (this is also why we had those classes as explicit strings everywhere: you can't easily tweak individual class strings if they're pulled from a template variable)

**Comment by kristinashu** [comment]
My feedback above all looks fixed. But looks like a bunch of vertical space was added between the `Click on a product to rate it.` text and the face. 

Screen cap from PR:
screen shot 2018-11-20 at 11 32 38 am

**Comment by mmmavis** [comment]
PR updated!

**Comment by Pomax** [comment]
Did we strip the white space on the sides intentionally? The homepage grid feels super crowded now

image

**Comment by kristinashu** [comment]
Let's see how wide desktop view looks on mofostaging with the real images.

Shouldn't be related but noticing the likelihood to buy filter isn't working properly (showing both thumbs up and down even though I've selected only up. Is that from this PR or a different one?

image

**Comment by mmmavis** [comment]
> Shouldn't be related but noticing the likelihood to buy filter isn't working properly (showing both thumbs up and down even though I've selected only up. Is that from this PR or a different one?

Might be me. I will check in a bit.

File: source/sass/buyers-guide/views/homepage.scss [hunk 2]
```diff
@@ -64,6 +64,11 @@
         padding: $padding-y $padding-x;
         width: calc((100% - #{$row-total-gutter-width}) / #{$items-per-row});
 
+        &.d-flex.d-none {
```
Comment (Pomax) [inline source/sass/buyers-guide/views/homepage.scss]: wow. crazy O_O

**Pull request merged**